extensions/ql-vscode/test/vscode-tests/test-config.ts

```diff
@@ -1,126 +1,225 @@
+import {
+  ConfigurationScope,
+  ConfigurationTarget,
+  workspace,
+  WorkspaceConfiguration as VSCodeWorkspaceConfiguration,
+} from "vscode";
 import { readFileSync } from "fs-extra";
 import { join } from "path";
-import { ConfigurationTarget } from "vscode";
-import { ALL_SETTINGS, InspectionResult, Setting } from "../../src/config";
 
-class TestSetting<T> {
-  private initialSettingState: InspectionResult<T> | undefined;
+function getIn(object: any, path: string): any {
+  const parts = path.split(".");
+  let current = object;
+  for (const part of parts) {
+    current = current[part];
+    if (current === undefined) {
+      return undefined;
+    }
+  }
+  return current;
+}
+
+function setIn(object: any, path: string, value: any): void {
+  const parts = path.split(".");
+  let current = object;
+  for (const part of parts.slice(0, -1)) {
+    if (current[part] === undefined) {
+      current[part] = {};
+    }
+    current = current[part];
+  }
+  current[parts[parts.length - 1]] = value;
+}
+
+interface WorkspaceConfiguration {
+  scope?: ConfigurationTarget;
+
+  get<T>(section: string | undefined, key: string): T | undefined;
+  has(section: string | undefined, key: string): boolean;
+  update(section: string | undefined, key: string, value: unknown): void;
```

Comment:
Should be consistent with the vscode API that this is mocking.

```suggestion
  update(section: string | undefined, key: string, value: unknown): Thenable<void>;
```

Reply:
I don't really think it needs to be consistent. This API is already inconsistent with the VSCode API because it's missing the `configurationTarget` and `overrideInLanguage` arguments. The [actual mocked method](https://github.com/github/vscode-codeql/blob/cac0add7887a6caaab8bd2329e52a2b32cc923b1/extensions/ql-vscode/test/vscode-tests/test-config.ts#L193) itself is `async` and I don't think we gain anything from also making this `async`.